src/main/java/com/gdsc/pikpet/config/security/UserDetailService.java

```diff
@@ -0,0 +1,28 @@
+package com.gdsc.pikpet.config.security;
+
+import com.gdsc.pikpet.repository.UserAccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class UserDetailService implements UserDetailsService {
+    private final UserAccountRepository userAccountRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserDetailService(UserAccountRepository userAccountRepository,PasswordEncoder passwordEncoder){
+        this.userAccountRepository = userAccountRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
```

Comment:
```suggestion
@Transactional
@RequiredArgsConstructor
public class UserDetailService implements UserDetailsService {
    private final UserAccountRepository userAccountRepository;
    private final PasswordEncoder passwordEncoder;
```
생성자 어노테이션으로 대체해서 주입하는게 깔끔할 거 같아요~

Reply:
굿~